### PR TITLE
Added 1.12 blocks

### PR DESCRIPTION
**Pull request by Bond-009**
I didn't test any of these changes because I'm waiting for #3757 

For a list of additions in 1.12: http://minecraft.gamepedia.com/1.12

### CODE REVIEW

**Comment by mathiascode** [comment]
The ID for concrete powder should be 252 in items.ini. Can you also add the new noteblock sounds in NoteEntity.cpp, and crafting recipes for colored beds?

**Comment by Bond-009** [comment]
I have exams, so I don't have a lot of time, but I will see what I can do.
And is the instrument enum in a certain order?

**Comment by Pokechu22** [comment]
Instrument values are at [Block Actions](http://wiki.vg/Block_Actions#Note_Block); the numbers there are significant.

File: src/BlockID.h [hunk 1]
```diff
@@ -252,12 +252,30 @@ enum BLOCKTYPE
...
+	E_BLOCK_BLACK_GLAZED_TERRACOTTA = 250,
+	E_BLOCK_CONCRETE = 251,
+	E_BLOCK_CONCRETE_POWDER = 252,
 	// ...
 	E_BLOCK_STRUCTURE_BLOCK = 255,
 
 	// Keep these two as the last values. Update the last block value to the last block with an id less than 255 when adding another block
 	// IsValidBlock() depends on this (255 gets checked additionally because there is a gap. See http://minecraft.gamepedia.com/Data_values#Block_IDs
-	E_BLOCK_NUMBER_OF_TYPES = E_BLOCK_BLACK_SHULKER_BOX + 1,  ///< Number of individual (different) blocktypes
+	E_BLOCK_NUMBER_OF_TYPES = E_BLOCK_CONCRETE + 1,  ///< Number of individual (different) blocktypes
```
Comment (madmaxoft) [inline src/BlockID.h]: Should be `E_BLOCK_CONCRETE_POWDER`

**Comment by Bond-009** [comment]
Do I need to change anything else?

**Comment by madmaxoft** [comment]
CI builds are still failing because you didn't add the documentation for some of the new constants in Lua API.

**Comment by Seadragon91** [comment]
I think we should point to the folder APIDump. Looking at the current a new contributor would have no clue where to add that :)

**Comment by madmaxoft** [comment]
I see we're slowly but steadily approaching the limit of our `BLOCKTYPE` type. We still use 8-bit block types everywhere, we'll really need to fix this soon.

**Pull request merged**